src/CMakeLists.txt

```diff
@@ -110,23 +126,14 @@ if(ENABLE_SWIFT)
                       -I ${PROJECT_SOURCE_DIR}
                       -I/usr/include
                       ${swift_optimization_flags}
+                    LINK_FLAGS
+                      $<TARGET_OBJECTS:DispatchStubs>
                     DEPENDS
-                      ${PROJECT_SOURCE_DIR}/dispatch/module.modulemap)
-
-  get_filename_component(swift_toolchain ${CMAKE_SWIFT_COMPILER} DIRECTORY)
-  get_filename_component(swift_toolchain ${swift_toolchain} DIRECTORY)
-  set(swift_runtime_libdir ${swift_toolchain}/lib/${swift_dir}/${swift_os}/${swift_arch})
-
-  target_sources(dispatch
-                 PRIVATE
-                   swift/DispatchStubs.cc
-                   ${CMAKE_CURRENT_BINARY_DIR}/swiftDispatch.o
-                   ${swift_runtime_libdir}/swiftrt.o)
-  if(CMAKE_BUILD_TYPE MATCHES Debug)
-    target_link_libraries(dispatch
-                          PRIVATE
-                            swiftSwiftOnoneSupport)
-  endif()
+                      ${PROJECT_SOURCE_DIR}/dispatch/module.modulemap
+                      module-map-symlinks
+                      DispatchStubs
+                    TARGET_FILES
+                      swiftDispatch_TARGET_FILES)
```

Comment:
TARGET_FILES isn't an argument to add_swift_library. I also noticed that you aren't passing it down to anything (since add_library is not passed the unmatched arguments from cmake parse arguments). What are you doing with this.

Reply:
Seems that I missed some of the changes.  It was meant to be an output parameter with the files that are generated (the library, the .swiftmodule, and the .swiftdoc) so that installation can be easier.  However, I realized that I cannot do that just yet (until the other changes for moving the libraries into the architecture specific directory is in).